src/app/public/modules/file-attachment/file-drop-control.component.spec.ts

```diff
@@ -0,0 +1,2957 @@
+import {
+  TestBed,
+  ComponentFixture,
+  async,
+  fakeAsync,
+  tick
+} from '@angular/core/testing';
+
+import {
+  By
+} from '@angular/platform-browser';
+
+import {
+  expect,
+  SkyAppTestUtility
+} from '@skyux-sdk/testing';
+
+import {
+  SkyFileDropControlComponent
+} from './file-drop-control.component';
+
+import {
+  SkyFileItem
+} from './file-item';
+
+import {
+  SkyFileLink
+} from './file-link';
+
+import {
+  FileDropControlTestModule
+} from './fixtures/file-drop-control.module.fixture';
+
+import {
+  FileDropControlTestComponent
+} from './fixtures/file-drop-control.compoennt.fixture';
+
+import {
+  FileDropControlNoFormTestComponent
+} from './fixtures/file-drop-control-no-form.component.fixture';
+
+import {
+  FileDropControlTemplateTestComponent
+} from './fixtures/file-drop-control-template.component.fixture';
+
+import {
+  FileDropTestHelper
+} from './fixtures/file-drop-test-helper';
+
+import {
+  SkyFileDropChange
+} from './types/file-drop-change';
+
+describe('File drop control component', () => {
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        FileDropControlTestModule
+      ]
+    });
+  });
+
+  /**
+   * NOTE: Most tests are the same as the file drop on 4/1/2020.
+   * This is to ensure the wrapper does not break existing functionality.
+   * Extra testing has been added to ensure that form controls work correctly
+   */
+
+  describe('reactive forms', () => {
```

Comment:
I'm not seeing any tests in here for checking the form control's state (dirty, valid, etc...). Should we also test for properly handling `required` and `disabled` states?

Reply:
1. I'll add some for checking the state

2. We don't yet fully support those. We have an issue for that but need design before we can fully implement it.